### PR TITLE
PCR Redeemed has issue with certificate screenflow

### PR DESCRIPTION
**Pull request by freshking**
## Description

Minor logic fix to secondary button deciding when it should be disabled.

## Link to Jira

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7892

### CODE REVIEW

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift [hunk 1]
```diff
@@ -64,7 +64,7 @@ class ExposureSubmissionTestCertificateViewController: DynamicTableViewControlle
 			didTapSecondaryButton { [weak self] isLoading in
 				guard let self = self else { return }
 
-				self.footerView?.setLoadingIndicator(false, disable: isLoading ? true : self.viewModel.isPrimaryButtonEnabled, button: .primary)
+				self.footerView?.setLoadingIndicator(false, disable: isLoading ? true : !self.viewModel.isPrimaryButtonEnabled, button: .primary)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift:67]: I don't understand how this fix is resolving the issue. The state of `isPrimaryButtonEnabled` is not changed based on the test result only after the birthday changes.

**Review by ArturFriesen — DISMISSED** [review]

**Comment by ArturFriesen** [comment]
Link to Jira is wrong.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift (hunk 1 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift:67]: Yes but this code is only called when selecting the secondary button. And then the state of the primary button is updated and it is incorrect that is should be disabled if `isPrimaryButtonEnabled` == true

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift:67]: So the logic is not depending in some other state, like testresult, but only if `isPrimaryButtonEnabled` was enabled, it should we disabled after loading? What about errors which could happen and after the error, the button should be still enabled for retry?

Comment (freshking) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift:67]: This fix is handling the error state. The user tries to register a redeemed PCR test and gets an error popup. Check out the branch and try it out.

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestCertificate/ExposureSubmissionTestCertificateViewController.swift:67]: You are right! I was confused by the other wrong code line. Which you have also fixed. Thanks!

**Review by ArturFriesen — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**